// ngx-pipes/esm5/src/pipes/string/a-or-an.js
import * as tslib_1 from "tslib";
import { Pipe } from '@angular/core';
import { isVowel } from '../helpers/helpers';
var AorAnPipe = /** @class */ (function () {
    function AorAnPipe() {
        this.irregularMap = {
            herb: 'an',
            honor: 'an',
            honorable: 'an',
            hour: 'an',
            mba: 'an',
            msc: 'an',
            'm.sc.': 'an',
            unicorn: 'a',
        };
    }
    AorAnPipe.prototype.transform = function (stringEntity) {
        if (!stringEntity || stringEntity === '') {
            return '';
        }
        else {
            var firstWord = stringEntity.trim().split(' ')[0];
            if (this.irregularMap[firstWord.toLocaleLowerCase()]) {
                return this.irregularMap[firstWord.toLocaleLowerCase()] + " " + stringEntity;
            }
            else {
                return isVowel(stringEntity[0]) ? "an " + stringEntity : "a " + stringEntity;
            }
        }
    };
    AorAnPipe = tslib_1.__decorate([
        Pipe({
            name: 'aOrAn',
        })
        /**
         * Takes a string and returns the string prepended by 'a' or 'an'.
         * Uses both naive and holdout-list approaches.
         * @constructor
         * @param {string} stringEntity - Entity to prepend 'a' or 'an' to.
         */
    ], AorAnPipe);
    return AorAnPipe;
}());
export { AorAnPipe };
